### PR TITLE
SV-925 Approvals Standard Cache

### PR DESCRIPTION
**Pull request by CodeWolf86**
PR to:
- Remove the Standard Options endpoint, It's defunct due to the implementation of the export endpoint under /ops
- Change Version from decimal to string. However we store it, decimal or string is an issue (1.10 == 1.1 in decimal format for example) So version has been changed to a string to store the exact value from Ifate.
- In addition, to account for the failings in storing as string or decimal in terms of precision loss or ordering, we have mirrored Ifates version storage. Ifate store 2 values, a major and minor, then join them when published to their Api. This has been mimicked allowing ordering by major then minor for correct versioning ordering.

### CODE REVIEW

**Review by ben1stone — CHANGES_REQUESTED** [review]

File: src/SFA.DAS.Courses.Data/Configuration/Standard.cs [hunk 1]
```diff
@@ -18,7 +18,7 @@ public void Configure(EntityTypeBuilder<Domain.Entities.Standard> builder)
             builder.Property(x => x.Title).HasColumnName("Title").HasColumnType("varchar").HasMaxLength(1000).IsRequired();
             builder.Property(x => x.IntegratedDegree).HasColumnName("IntegratedDegree").HasColumnType("varchar").HasMaxLength(100).IsRequired();
             builder.Property(x => x.Level).HasColumnName("Level").HasColumnType("int").IsRequired();
-            builder.Property(x => x.Version).HasColumnName("Version").HasColumnType("decimal");
+            builder.Property(x => x.Version).HasColumnName("Version").HasColumnType("varchar").HasMaxLength(50);
```
[the remark this answers is not in the document]
Comment (CodeWolf86) [reply src/SFA.DAS.Courses.Data/Configuration/Standard.cs:21]: Yep it should. Changed

File: src/SFA.DAS.Courses.Data/Extensions/StandardFilterExtension.cs [hunk 2]
```diff
@@ -51,7 +51,7 @@ public static IEnumerable<Standard> InMemoryFilterIsLatestVersion(this IEnumerab
                         // To include all of the above we need to create a list that is grouped by LarsCode (otherwise unnecessary)
                         // and union it with the `actual` list which is grouped by IFateReferenceNumber 
 
-                        var standardsByLarsCode = standards.GroupBy(s => s.LarsCode).Select(c => c.OrderByDescending(x => x.Version).FirstOrDefault());
+                        var standardsByLarsCode = standards.GroupBy(s => s.LarsCode).Select(c => c.OrderByDescending(x => x.VersionMajor).ThenByDescending(y => y.VersionMinor).FirstOrDefault());
 
                         var standardsByIfateReferenceNumber = standards.GroupBy(s => s.IfateReferenceNumber).Select(c => c.OrderByDescending(x => x.Version).FirstOrDefault());
```
[the remark this answers is not in the document]
Comment (CodeWolf86) [reply src/SFA.DAS.Courses.Data/Extensions/StandardFilterExtension.cs:56]: Yes for consistency. Changed.

File: src/SFA.DAS.Courses.Domain/Extensions/ToStandardVersionIdExtension.cs [hunk 3]
```diff
@@ -2,10 +2,10 @@
 {
     public static class ToStandardVersionIdExtension
     {
-        public static string ToStandardVersionId(this string ifateReferenceNumber, decimal? version)
+        public static string ToStandardVersionId(this string ifateReferenceNumber, string version)
```
[the remark this answers is not in the document]
Comment (CodeWolf86) [reply src/SFA.DAS.Courses.Domain/Extensions/ToStandardVersionIdExtension.cs:5]: Yep have renamed it.

**Review by ben1stone — APPROVED** [review]

File: src/SFA.DAS.Courses.Domain/Courses/Standard.cs [hunk 4]
```diff
@@ -12,7 +12,7 @@ public class Standard
         public string Status { get; set; }
         public string Title { get; set; }
         public int Level { get; set; }
-        public decimal Version { get; set; }
+        public string Version { get; set; }
```
Comment (StuartReeve102) [inline src/SFA.DAS.Courses.Domain/Courses/Standard.cs:15]: This refers to multiple classes, where you have added a string Version and interger VersionMajor and VersionMinor, properties, shouldn't the string Version property be a calculated property based on VersionMajor and VersionMinor?

Comment (StuartReeve102) [inline src/SFA.DAS.Courses.Domain/Courses/Standard.cs:15]: This refers to multiple classes, where you have added a string Version and interger VersionMajor and VersionMinor, properties, shouldn't the string Version property be a calculated property based on VersionMajor and VersionMinor?

**Review by StuartReeve102 — CHANGES_REQUESTED** [review]

File: src/SFA.DAS.Courses.Domain/Courses/Standard.cs (hunk 4 above)
Comment (CodeWolf86) [reply src/SFA.DAS.Courses.Domain/Courses/Standard.cs:15]: Given the data from Ifate comes in as a string, it seemed best to store that as is so we hold the original value. The VersionMajor VersionMinor is the calculated aspect from splitting the original field. The sole purpose of the split was purely for ordering purposes at the database level. It is a little bit duplicated but it seemed the simplest option.

**Review by StuartReeve102 — APPROVED** [review]

**Pull request merged**